core/pkg/server/stream_init.go

```diff
@@ -43,6 +44,23 @@ func NewBackend(
 	})
 }
 
+func ProxyFn(proxyMap map[string]string) func(*http.Request) (*url.URL, error) {
```

Comment:
doc comment needed

* short summary sentence
* what are the expected contents of `proxyMap`
* what does this return
* why is `http.ProxyFromEnvironment` insufficient